doc/source/rllib/rllib-catalogs.rst

```diff
@@ -0,0 +1,178 @@
+An :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` creates an instance of the Catalog class they receive as part of their constructor.
+It then create its internal ``models`` and ``distributions`` with the help of this Catalog.
+
+.. note::
+    You can also modify :py:class:`~ray.rllib.core.models.Model` or :py:class:`~ray.rllib.models.distributions.Distribution` in an :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule` directly by overriding the RLModule's constructor!
+
+.. image:: images/catalog/catalog_and_rlm_diagram.svg
+    :align: center
+
+The following diagram shows a concrete case in more detail.
+
+.. dropdown:: **Example of catalog in a PPORLModule**
+    :animate: fade-in-slide-down
+
+    The :py:class:`~ray.rllib.algorithms.ppo.ppo_catalog.PPOCatalog` is fed an `observation space``, ``action space``,
+    a ``model config dict`` and the ``view requirements`` of the :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`.
+    The model config dicts and the view requirements are only of interest in special cases, such as
+    recurrent networks or attention networks. A PPORLModule has four components that are created by the
+    :py:class:`~ray.rllib.algorithms.ppo.ppo_catalog.PPOCatalog`:
+    ``Encoder``, ``value function head``, ``policy head`` and ``action distribution``. You can find out more about this
+    distinction between these components in our section on Models.
+
+    .. image:: images/catalog/ppo_catalog_and_rlm_diagram.svg
+        :align: center
+
+
+Inject your custom models into RLModules
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can make a :py:class:`~ray.rllib.core.models.catalog.Catalog` build custom ``models`` by overriding the Catalog’s methods used by RL Modules to build ``models``.
+Have a look at these lines from the constructor of the :py:class:`~ray.rllib.algorithms.ppo.ppo_torch_rl_module.PPOTorchRLModule` to see how Catalogs are being used by an :py:class:`~ray.rllib.core.rl_module.rl_module.RLModule`:
+
+.. literalinclude:: ../../../rllib/algorithms/ppo/ppo_base_rl_module.py
+    :language: python
+    :start-after: __sphinx_doc_begin__
+    :end-before: __sphinx_doc_end__
+
+Consequently, in order to build a custom :py:class:`~ray.rllib.core.models.Model` compatible with a PPORLModule,
+you can override methods by inheriting from :py:class:`~ray.rllib.algorithms.ppo.ppo_catalog.PPOCatalog`
+or write a :py:class:`~ray.rllib.core.models.catalog.Catalog` that implements them from scratch.
+The following showcases such modifications.
+
+This example shows two things:
+
+- How to write a custom :py:class:`~ray.rllib.models.distributions.Distribution`
+- How to inject a custom action distribution into a :py:class:`~ray.rllib.core.models.catalog.Catalog`
+
+.. literalinclude:: ../../../rllib/examples/catalog/custom_action_distribution.py
+   :language: python
+   :start-after: __sphinx_doc_begin__
+   :end-before: __sphinx_doc_end__
+
+
+
+Notable TODOs
```

Comment:
Do you want to hide this list?

Reply:
It's not being hidden in other docs we have, but I think it's actually a good idea to hide it.

Reply:
CC @kouroshHakha